tests/test_rest_api/test_limsdb/test_limsdb.py

```diff
@@ -0,0 +1,481 @@
+                       'processes':
+                           [
+                               {'date': 'Jun 01 2015',
+                                'process_id': None,
+                                'type': 'complete',
+                                'name': 'Receive Sample EG 6.1'}
+                           ],
+                       'name': 'sample_submission'},
+                      {'date': 'Jul 16 2015',
+                       'processes':
+                           [
+                               {'date': 'Jul 16 2015',
+                                'process_id': None,
+                                'type': 'complete',
+                                'name': 'Read and Eval SSQC'}
+                           ],
+                       'name': 'sample_qc'},
+               ]
+
+        self.sample1.add_queue_location('Sequencing Plate Preparation EG 2.0', datetime.strptime('20-07-15', '%d-%m-%y'))
+        all_status = self.sample1.all_statuses()
+        assert len(all_status) == 2
+        self.sample1._all_statuses_and_date = None
+        all_status = self.sample1.all_statuses()
+        assert len(all_status) == 3
+        assert all_status == \
+               [
+                      {'date': 'Jun 01 2015',
+                       'processes':
+                           [
+                               {'date': 'Jun 01 2015',
+                            'process_id': None,
+                                'type': 'complete',
+                                'name': 'Receive Sample EG 6.1'}
+                           ],
+                       'name': 'sample_submission'},
+                      {'date': 'Jul 16 2015',
+                       'processes':
+                           [
+                               {'date': 'Jul 16 2015',
+                                'process_id': None,
+                                'type': 'complete',
+                                'name': 'Read and Eval SSQC'}
+                           ],
+                       'name': 'sample_qc'},
+                      {'date': 'Jul 20 2015',
+                       'processes':
+                           [
+                               {'date': 'Jul 20 2015',
+                                'process_id': None,
+                                'type': 'queued',
+                                'name': 'Sequencing Plate Preparation EG 2.0'}
+                           ],
+                       'name': 'library_queue'}
+               ]
+
+        self.sample1._processes.remove(('Sequencing Plate Preparation EG 2.0', datetime(2015, 7, 20, 0, 0), 'queued', None))
+
+    def test_get_status_and_date(self):
+        status_and_date = self.sample1._get_status_and_date()
+        assert status_and_date == ('sample_qc', datetime(2015, 6, 1, 0, 0))
+        self.sample1.add_queue_location('Sequencing Plate Preparation EG 2.0', datetime.strptime('20-07-15', '%d-%m-%y'))
+        self.sample1._status_and_date = None
```

Comment:
The fact that you have to reset `_status_and_date` here tells me that it should happen in `add_completed_process` and  `add_queue_location`. 
Can you reset `_status_and_date` and `_all_statuses_and_date` to `None` in both `add_completed_process` and  `add_queue_location`?